NotificationApi/NotificationApi/Controllers/NotificationController.cs

```diff
@@ -0,0 +1,103 @@
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using NotificationApi.Common;
+using NotificationApi.Contract.Requests;
+using NotificationApi.Contract.Responses;
+using NotificationApi.DAL.Commands;
+using NotificationApi.DAL.Commands.Core;
+using NotificationApi.DAL.Queries;
+using NotificationApi.DAL.Queries.Core;
+using NotificationApi.Domain;
+using NotificationApi.Domain.Enums;
+using NotificationApi.Extensions;
+using Notify.Interfaces;
+using NSwag.Annotations;
+
+namespace NotificationApi.Controllers
+{
+    [Produces("application/json")]
+    [Route("Notification")]
+    [ApiController]
+    public class NotificationController : ControllerBase
+    {
+        private readonly IQueryHandler _queryHandler;
+        private readonly IAsyncNotificationClient _asyncNotificationClient;
+        private readonly ICommandHandler _commandHandler;
+
+        public NotificationController(IQueryHandler queryHandler, IAsyncNotificationClient asyncNotificationClient,
+            ICommandHandler commandHandler)
+        {
+            _queryHandler = queryHandler;
+            _asyncNotificationClient = asyncNotificationClient;
+            _commandHandler = commandHandler;
+        }
+
+        [HttpGet("template/{notificationType}")]
+        [OpenApiOperation("GetTemplateByNotificationTypeAsync")]
+        [ProducesResponseType(typeof(NotificationTemplateResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> GetTemplateByNotificationTypeAsync(int notificationType)
+        {
+            var template = await _queryHandler.Handle<GetTemplateByNotificationTypeQuery, Template>(new GetTemplateByNotificationTypeQuery((NotificationType)notificationType));
+            if (template == null)
+            {
+                throw new BadRequestException($"Invalid {nameof(notificationType)}: {notificationType}");
+            }
+
+            return Ok(new NotificationTemplateResponse
+            {
+                Id = template.Id,
+                NotificationType = (int)template.NotificationType,
+                NotifyemplateId = template.NotifyTemplateId,
+                Parameters = template.Parameters
+            });
+        }
+
+        [HttpPost]
+        [OpenApiOperation("CreateNewNotificationAsync")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> CreateNewNotificationAsync(AddNotificationRequest request)
+        {
+            var template = await _queryHandler.Handle<GetTemplateByNotificationTypeQuery, Template>(new GetTemplateByNotificationTypeQuery((NotificationType)request.NotificationType));
```

Comment:
This is the beefiest of all the endpoints. Maybe wrap this into a service class to handle all the heavy lifting